ykllvmwrap/src/memman.cc

```diff
@@ -0,0 +1,78 @@
+#include <llvm/ExecutionEngine/RTDyldMemoryManager.h>
+#include <sys/mman.h>
+
+using namespace llvm;
+
+struct AllocMem {
+  uint8_t *Ptr;
+  uintptr_t Size;
+};
+
+class MemMan : public RTDyldMemoryManager {
+public:
+  MemMan();
+  ~MemMan() override;
+
+  uint8_t *allocateCodeSection(uintptr_t Size, unsigned Alignment,
+                               unsigned SectionID,
+                               StringRef SectionName) override;
+
+  uint8_t *allocateDataSection(uintptr_t Size, unsigned Alignment,
+                               unsigned SectionID, StringRef SectionName,
+                               bool isReadOnly) override;
+
+  bool finalizeMemory(std::string *ErrMsg) override;
+  void freeMemory();
+
+private:
+  std::vector<AllocMem> code;
+  std::vector<AllocMem> data;
+};
+
+MemMan::MemMan() {}
+MemMan::~MemMan() {}
+
+uint8_t *alloc_mem(uintptr_t Size, unsigned Alignment,
+                   std::vector<AllocMem> *Vec) {
+  uintptr_t RequiredSize = Alignment * ((Size + Alignment - 1) / Alignment + 1);
```

Comment:
Odd indent?

Reply:
Or maybe the formatter is using 2 spaces? Which do we want?

Reply:
Yeah, C++ uses 2 spaces instead of 4 according to the style guide. We can change, it maybe sticking to what `clang-format` does by default causes us less grief in the long term.

Reply:
I like to stick to defaults so that we don't have to configure anything.